src/tp394and395.jl

```diff
@@ -0,0 +1,31 @@
+# TP problem 394 and 395 in NLS format
+#
+#   Source:
+#   Klaus Schittkowski,
+#   More test examples for nonlinear programming codes,
+#   Lecture Notes in Economics and Mathematical Systems 282,
+#   Springer Verlag Berlin Heidelberg, 1987
+#   10.1007/978-3-642-61582-5
+#
+# A. Montoison, Montréal, 06/2018.
+
+export tp394, tp395
+
+function tp394and395(n :: Int=20; version :: String="tp394")
+
+  nls = Model()
+  @variable(nls, x[i=1:n], start=2)
+
+  @NLexpression(nls, FA[i=1:n], sqrt(i) * x[i])
+  @NLexpression(nls, FB[i=1:n], sqrt(i) * x[i]^2)
+
+  @NLconstraint(nls, sum(x[i]^2 for i=1:n) == 1)
+
+  return MathProgNLSModel(nls, [FA; FB], name=version)
+end
+
+"Test problem 394 in NLS format"
+tp394(n :: Int=20) = tp394and395(n, version="tp394")
```

Comment:
I'm not sure if having this argument is good, because the problem below would be the same as this. Perhaps in the future, we can export the more general problem as well.
@dpo, what do you think?

Reply:
I didn't want to keep this argument... I begin to implement similar problems like [```dixmaan_efgh.jl```](https://github.com/JuliaSmoothOptimizers/OptimizationProblems.jl/blob/master/src/dixmaan_efgh.jl) in OptimizationProblems.jl repository. Where one problem take parameters and the other ones call it.

Then I saw your commentary, where you use an auxiliary function ```tpxxxtoxxx``` and where each problem don't take argument. So, all problems ```tp*.jl``` are implemented in the same way without argument.
It's why in that file we have I mix of the two possibilities.